### DISCUSSION TITLE
Cant able to open entire codebase

### DISCUSSION THREAD

**Discussion by leo-faizan** [zed-industries/zed#12432] [Feature Requests]
- [ ] Completed

I am a linux user trying to switch from vscode to zed

I don't see an option to `open folder` instead I am only able to `open` a single file. Obvious reasons this becoming tedious when working with huge codebases. 

![image](https://github.com/zed-industries/zed/assets/65481389/b765ef7c-ed35-4dc4-b3f2-3eec650277ad)

### Environment

cargo run --release

**Comment by leo-faizan** [comment]
I am new to setting up IDEs and stuff but very much interesting. Please guide me if I am missing out on something silly

**Comment by CharlesChen0823** [comment]
current using `xdg-desktop-portal` to open directory, you might need update or install this.

**Comment by leo-faizan** [comment]
> current using `xdg-desktop-portal` to open directory, you might need update or install this.

could you be a little more specific on what to do?

**Comment by CharlesChen0823** [comment]
in ubuntu, you can install [xdg-desktop-portal](https://installati.one/install-xdg-desktop-portal-ubuntu-20-04/)

**Comment by jansol** [comment]
More importantly, you need to install a [backend](https://wiki.archlinux.org/title/XDG_Desktop_Portal#List_of_backends_and_interfaces) that actually implements the file chooser function. Generally you'll want the one that matches your desktop environment most closely. If none does, the gtk one is a popular fallback.

**Comment by leo-faizan** [comment]
> More importantly, you need to install a [backend](https://wiki.archlinux.org/title/XDG_Desktop_Portal#List_of_backends_and_interfaces) that actually implements the file chooser function. Generally you'll want the one that matches your desktop environment most closely. If none does, the gtk one is a popular fallback.

Thanks for your response.

I have installed both `xdg-desktop-portal` and `xdg-desktop-portal-gtk`. Nothing has changed. Do I need to install multiple backends for this? I am helpless on how to proceed.

**Comment by jansol** [comment]
You need to make sure the portal service actually runs. With GNOME and Plasma this should be handled by the DE AFAIK? With others you may have to set it up manually (explained in the arch wiki page I linked above).

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
